src/page-handler.js

```diff
@@ -0,0 +1,680 @@
+        }
+
+        try {
+            const pageQs = await this._getPageQs();
+            const results = this.__getPageQsResults(pageQs);
+
+            shouldContinue = await this._processPageQsResults(results, queryZpid);
+
+            if (shouldContinue) {
+                const extracted = await this._extractQueryStatesForCurrentPage(pageQs);
+                queryStates.push(...extracted.states);
+                totalCount = extracted.totalCount;
+            }
+        } catch (/** @type {any} */ e) {
+            log.debug(e);
+        }
+
+        log.debug('searchState', { queryStates });
+
+        if (shouldContinue && queryStates?.length) {
+            await this._processExtractedQueryStates(queryStates, totalCount, queryZpid);
+        }
+    }
+
+    /**
+     *
+     * @param {string} zpid
+     * @param {string} detailUrl
+     * @param {ReturnType<typeof createQueryZpid>} queryZpid
+     * @returns
+     */
+    async processZpid(zpid, detailUrl, queryZpid) {
+        const { page, request, requestQueue, session } = this.context;
+        const { zpids } = this.globalContext;
+
+        if (this.isOverItems()) {
+            return;
+        }
+
+        try {
+            if (!zpid) {
+                throw new Error(`Zpid not string or number`);
+            }
+
+            if (+zpid != zpid) {
+                throw new Error('Invalid non-numeric zpid');
+            }
+
+            if (zpids.has(`${zpid}`)) {
+                log.debug(`Zpids already contain zpid ${zpid}, returning from process zpid`);
+                return;
+            }
+
+            if (!session.isUsable()) {
+                throw new Error('Not trying to retrieve data, session is not usable anymore');
+            }
+
+            log.debug(`Extracting ${zpid}`);
+
+            await this.extendOutputFunction(
+                JSON.parse(await queryZpid(page, zpid)).data.property,
+                {
+                    request,
+                    page,
+                    zpid,
+                },
+            );
+        } catch (e) {
+            this.anyErrors = true;
+            session.markBad();
+            log.debug('processZpid', { error: e });
+
+            if (this.isOverItems()) {
+                return;
+            }
+
+            // add as a separate detail for retrying
+            await requestQueue.addRequest({
+                url: new URL(detailUrl || `/homedetails/${zpid}_zpid/`, 'https://www.zillow.com').toString(),
+                userData: {
+                    label: LABELS.DETAIL,
+                    zpid: +zpid,
+                },
+            }, { forefront: true });
+        } finally {
+            await sleep(100);
+        }
+    }
+
+    isOverItems(extra = 0) {
+        const { zpids, input } = this.globalContext;
+        return (typeof input.maxItems === 'number' && input.maxItems > 0
+            ? (zpids.size + extra) >= input.maxItems
+            : false);
```

Comment:
You don't need all these braces. ES lint should capture that?

Reply:
It is not capturing :/. I reviewed it locally with modules installed and no luck.